fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -0,0 +1,365 @@
+
+            indexBuilder.buildIndexAsync(true)
+                    .handle((ignore, e) ->
+                    {
+                        assertNull(e);
+                        return null;
+                    })
+                    .join();
+        }
+        assertEquals(numRecords, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_SCANNED));
+        assertEquals(numRecords, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_INDEXED));
+    }
+
+    @Test
+    public void testIndexFromIndexNoFallback() {
+
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final long numRecords = 7;
+
+        List<TestRecords1Proto.MySimpleRecord> records = LongStream.range(0, numRecords).mapToObj(val ->
+                TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).build()
+        ).collect(Collectors.toList());
+        Index srcIndex = new Index("src_index", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index("tgt_index", field("num_value_3_indexed").ungrouped(), IndexTypes.SUM);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.addIndex("MySimpleRecord", srcIndex);
+            metaDataBuilder.addIndex("MySimpleRecord", tgtIndex);
+        } ;
+
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext())  {
+            records.forEach(recordStore::saveRecord);
+            context.commit();
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setDatabase(fdb).setMetaData(metaData).setIndex(srcIndex).setSubspace(subspace)
+                    .build()) {
+                recordStore.markIndexWriteOnly(srcIndex).join();
+                indexBuilder.rebuildIndex(recordStore);
+                recordStore.markIndexReadable(srcIndex).join();
+                context.commit();
+                assertTrue(recordStore.isIndexReadable(srcIndex));
+            }
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            recordStore.markIndexWriteOnly(tgtIndex).join();
+            context.commit();
+            assertTrue(recordStore.isIndexWriteOnly(tgtIndex));
+        }
+
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(tgtIndex).setSubspace(subspace)
+                .setIndexFromIndex(new OnlineIndexer.IndexFromIndexRequest("src_index")
+                        .mustUseSourceIndex())
+                .setTimer(timer)
+                .build()) {
+
+            indexBuilder.buildIndexAsync(true)
+                    .handle((ignore, e) ->
+                    {
+                        assertNotNull(e);
```

Comment:
I think this assert can be replaced by something like:

```
MetaDataException e = assertThrows(MetaDataException.class, indexBuilder::buildIndex);
```

And then you can assert on the error, etc.

Reply:
Nice!